packages/auth/test/integration/flows/totp.test.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ import { expect, use } from 'chai';
+ import chaiAsPromised from 'chai-as-promised';
+ import * as sinon from 'sinon';
+ import sinonChai from 'sinon-chai';
+ 
+//import { mockTotp } from '../../helpers/integration/helpers';
+import {Auth, createUserWithEmailAndPassword, multiFactor, signInAnonymously, signInWithEmailAndPassword, UserCredential} from '@firebase/auth';
+
+import {
+  cleanUpTestInstance,
+  getTestInstance,
+  mockTotp,
+  randomEmail
+} from '../../helpers/integration/helpers';
+import { MultiFactorAssertionImpl } from '../../../src/mfa/mfa_assertion';
+
+import { MultiFactorSessionImpl } from '../../../src/mfa/mfa_session';
+import { TotpMultiFactorAssertionImpl, TotpMultiFactorGenerator, TotpSecret } from '../../../src/mfa/assertions/totp';
+import * as MFA from '../../../src/api/account_management/mfa';
+import { FirebaseError } from '@firebase/util';
+
+
+
+use(chaiAsPromised);
+use(sinonChai);
+
+const TOTP_COMB_A = {
+
+    response: { sharedSecretKey: 'secretKey3',
+              verificationCodeLength: 30,
+              hashingAlgorithm: 'sha1',
+              periodSec:30,
+              sessionInfo: 'testsSessionInfo',
+              finalizeEnrollmentTime:  Date.now()
+              },
+    code: '...'
+  };
+  
+  const TOTP_COMB_B = {
+
+    response: { sharedSecretKey: 'secretKey2',
+    verificationCodeLength: 30,
+    hashingAlgorithm: 'sha1',
+    periodSec: 30,
+    sessionInfo: 'testsSessionInfo',
+    finalizeEnrollmentTime:  Date.now()
+    },
+    code: '...'
+  };
+
+describe(' Integration tests: Mfa TOTP', () => {
+    let auth: Auth;
+    let idToken: string;
+    let signUpCred: UserCredential;
+    let email: string;
+    let assertion: MultiFactorAssertionImpl;
+    let _request: MFA.StartTotpMfaEnrollmentRequest;
+    let startMfaResponse: MFA.StartTotpMfaEnrollmentResponse;
+    let displayName: string;
+  beforeEach(async () => {
+    auth = getTestInstance();
```

Comment:
We don't need a new email/user per test case. might be enough to initialize just auth and email in the beforeEach block here.